src/smart-components/myUserAccess/MUARolesTable.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import debounce from 'lodash/debounce';
+import { TableToolbarView } from '../../presentational-components/shared/table-toolbar-view';
+import { fetchRoles, fetchRoleForPrincipal } from '../../redux/actions/role-actions';
+import { ListLoader } from '../../presentational-components/shared/loader-placeholders';
+
+import {
+    Table,
+    TableHeader,
+    TableBody,
+    TableVariant,
+    compoundExpand,
+    cellWidth
+} from '@patternfly/react-table';
+
+const columns = [
+    'Roles',
+    'Description',
+    {
+        title: 'Permissions',
+        cellTransforms: [ compoundExpand, cellWidth(20) ]
+    }
+];
+
+let debouncedFetch;
+
+const MUARolesTable = ({
+    fetchRoles,
+    fetchRoleForPrincipal,
+    roles,
+    isLoading,
+    rolesWithAccess
+}) => {
+
+    const [ filter, setFilter ] = useState('');
+    const [ expanded, setExpanded ] = useState({});
+
+    const createRows = (data) => {
+
+        return data ? data.reduce((acc, { uuid, name, description, accessCount }, i) => ([
+            ...acc, {
+                uuid,
+                cells: [{ title: name, props: { component: 'th', isOpen: false }},
+                { title: description, props: { isOpen: false }},
+                { title: accessCount, props: { isOpen: expanded[uuid] === 2 }}
+                ]
+            },
+            {
+                uuid: `${uuid}-groups`,
+                parent: 2 * i,
+                compoundParent: 2,
+                cells: [{
+                    props: { colSpan: 4, className: 'pf-m-no-padding' },
+                    title: rolesWithAccess && rolesWithAccess[uuid]
+                        ? <Table
+                            aria-label="Simple Table"
+                            borders={ false }
+                            variant={ TableVariant.compact }
+                            cells={ [ 'Application', 'Resource type', 'Operation' ] }
+                            rows={ rolesWithAccess[uuid].access.map(access => ({ cells: access.permission.split(':') })) }>
+                            <TableHeader />
+                            <TableBody />
+                        </Table>
+                        : <ListLoader />
+
+                }]
+            }
+        ]), []) : [];
+    };
+
+    useEffect(() => {
+        fetchRoles({ limit: 20, offset: 0, scope: 'principal' });
+        debouncedFetch = debounce(
```

Comment:
It's better to use `useCallback` instead of `useEffect` hook, as you don't know from the variable when it gets created (alwo, shouldn't that be 800 instead of 500 milis?)
```JS
const debouncedFetch = useCallback(debounce(
  (limit, offset, name, addFields) => fetchRoles({ limit, offset, name, addFields }),
  800
), [])
```